### DISCUSSION TITLE
Welcome to opentelemetry-erlang-contrib Discussions!

### DISCUSSION THREAD

**Discussion by bryannaegele** [open-telemetry/opentelemetry-erlang-contrib#7] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by himadrica** [comment]
Hi, I am Himadri. I am a software developer. Currently learning Erlang and I have a project about instrumenting Erlang app with opentelemtry. So my design is a java application and an Erlang app taking each other with http call. I am making a http request to the java rest endpoint but I am getting two separate traces instead of 1 trace between Erlang and Java. Here is my Erlang oepntelemetry configurations

sys.config
`[
  {opentelemetry,
  [{processors,
    [{otel_batch_processor,
        #{exporter => {opentelemetry_exporter, #{protocol => grpc,
                                                 endpoints => [{http, "localhost", 4317, []}]}}}}]}]}
  ,
  {text_map_propagators, [baggage,
                        trace_context]}
].
`

rebar.config

`{erl_opts, [debug_info]}.
{deps, [
    {opentelemetry_api, "~> 1.0"}, 
    {opentelemetry, "~> 1.0"},
    {amqp_client, "3.9.11"},
    {opentelemetry_exporter, {git_subdir, "http://github.com/open-telemetry/opentelemetry-erlang", {branch, "main"}, "apps/opentelemetry_exporter"}}
  ]}.

{relx, [
        {release, {otel_multi_node, "0.1.0"},
            [
              opentelemetry_exporter,
              {opentelemetry, temporary}, 
              otel_multi_node,
              sasl
            ]
          },

        {mode, dev},

        %% automatically picked up if the files
        %% exist but can be set manually, which
        %% is required if the names aren't exactly
        %% sys.config and vm.args
        {sys_config, "./config/sys.config"},
        {vm_args, "./config/vm.args"}

        %% the .src form of the configuration files do
        %% not require setting RELX_REPLACE_OS_VARS
        %% {sys_config_src, "./config/sys.config.src"},
        %% {vm_args_src, "./config/vm.args.src"}
]}.`

*.app.src
`{application, otel_multi_node,
 [{description, "An OTP application"},
  {vsn, "0.1.0"},
  {registered, []},
  {mod, {otel_multi_node_app, []}},
  {applications,
   [kernel,
    stdlib,
    opentelemetry_api,
    opentelemetry
   ]},
  {env,[]},
  {modules, []},

  {licenses, ["Apache 2.0"]},
  {links, []}
 ]}.
`

erlang function:
`httpCall() ->
    Attributes = #{
                  'http.client_ip' => 'localhost',
                  'http.flavor' => 'http',
                  'http.host' => 'localhost:8080',
                  'http.host.port' => '8080',
                  'http.method' => 'GET',
                  'http.scheme' => '',
                  'http.target' => 'producer',
                  'net.host.ip' => '127.0.0.1',
                  'net.transport' => 'IP.TCP'
                 },
    ?with_span(<<"http call to producer">>, Attributes,
        fun(_whatever) ->
            Method = get,
            URL = <<"http://localhost:8080/send?msg=hello">>,
            Headers = [],
            Payload = <<>>,
            Options = [],
            httpc:request(URL)
        end),
ok.
`

I am not sure why the context is not propagating. Am I missing any configuration or do I need some other instrumentation?

**Reply by tsloughter** [reply]
Hi @himadrica please make a separate discussion topic to discuss specific questions like this. But I can also note that for `httpc` you need to include the propagation headers yourself (`httpc` has no middleware/hook support). There is actually an example of just this in the `interop` tests: https://github.com/open-telemetry/opentelemetry-erlang/blob/main/interop/w3c_trace_context_interop.erl#L52-L59

**Reply by himadrica** [reply]
Thank you @tsloughter for your reply. I will create a separate topic.
